src/map/pc.c

```diff
@@ -2240,17 +2241,14 @@ static int pc_bonus_item_drop(struct s_add_drop *drop, const short max, short id
 		if (rate > -1)
 			rate = -1;
 	}
-	for(i = 0; i < max && (drop[i].id || drop[i].group); i++) {
-		if (((id && drop[i].id == id) || (group && drop[i].group == group))
-		 && race_mask != RCMASK_NONE
-		) {
+	for(i = 0; i < max && (drop[i].id != 0 || drop[i].is_group); i++) {
```

Comment:
while you changing this code, add space after for: ``for (i = ...``

Reply:
oh good catch. Will do so later.